api/client/src/main/java/org/projectnessie/client/http/NessieApiCompatibilityFilter.java

```
@@ -65,9 +67,14 @@ static void check(int clientApiVersion, HttpClient httpClient)
           e.toString());
       return;
     } catch (Exception e) {
-      LOGGER.warn(
-          "API compatibility check: failed to contact config endpoint, proceeding without check",
-          e);
+      if (hasCauseMatching(
+          e, t -> t instanceof InterruptedException || t instanceof TimeoutException)) {
+        Thread.currentThread().interrupt();
```

We still failed to validate API compatibility in this case... Why skip the WARN log?

---

The compatibility check _cannot_ succeed in these cases - it's aborted. No need to confuse users.

---

How do we know the interrupt actually kills the client? It may be used it customized environments?

---

That's hypothetical, no?

---

No. Interrupting a thread does not mean the client will not be reused.

---

I'd say, if we silently ignore the interrupt here, we ought to redo the API check on the next request (which currently will not happen).